kong/templates/nginx_kong.lua

```diff
@@ -440,12 +440,6 @@ server {
             Kong.serve_wrpc_listener()
         }
     }
-
-    location = /version-handshake {
-        content_by_lua_block {
-            Kong.serve_version_handshake()
-        }
-    }
```

Comment:
I am not opposed to removing this, but what is the context? Is this dead code? (It is ok to just answer and mark this as "Done").

Reply:
Yes, Now we use `HEAD` request to check protocol (#8834), the old version handshake is unnecessary, so we could remove it.

You could also refer to #8775, in this PR, we use wrpc to do version negotiation and remove this endpoint.